doc/source/dsintro.rst

```diff
@@ -507,24 +507,17 @@ of one argument to be called on the ``DataFrame``. A *copy* of the original
 DataFrame is returned, with the new values inserted.
 
 .. warning::
+  Starting from Python 3.6 ``**kwargs`` is an ordered dictionary and :func`DataFrame.assign`
+  respects the order of the keyword arguments. You may now use assign in the following way:
 
-  Since the function signature of ``assign`` is ``**kwargs``, a dictionary,
-  the order of the new columns in the resulting DataFrame cannot be guaranteed
-  to match the order you pass in. To make things predictable, items are inserted
-  alphabetically (by key) at the end of the DataFrame.
+  .. ipython:: python
 
-  All expressions are computed first, and then assigned. So you can't refer
-  to another column being assigned in the same call to ``assign``. For example:
+      In [1]: df.assign(C = lambda x: x['A'] + x['B'],
```

Comment:
when using an ipython:: python block, it executes the code, thus we cannot have the ``In [1]: `` there

Reply:
using a simple python:: block will always execute the code and yield errors since df is (currently) not defined appropriately. So - it seems - there are two possibilities to resolve this:
1. use verbatim and just show the syntax of the call (which you refused in my last commit)
2. use the iris DataFrame from the section or df from a section before and modify the call w.r.t those frames.

Which way to go?

Reply:
Call it something other than `df` though, since creating a new `df` will break examples further down.